veps/sig-compute/pcihole64-proposal.md

```diff
@@ -0,0 +1,323 @@
+# VEP #33: Allow to disable the 64-bit PCI hole
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone /
+release*.
+
+- [ ] (R) Enhancement issue created, which links to VEP dir
+  in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+<!--
+Provide a brief overview of the topic
+-->
+
+This proposal focuses on allowing to disable the 64-bit PCI hole in
+`VirtualMachines`, so that legacy OSes like Windows XP or Server 2k3 can run on
+KubeVirt.
+
+## Motivation
+
+<!--
+Why this enhancement is important
+-->
+
+At the moment legacy OSes like Windows XP or Server 2k3 do not work out of the
+box on KubeVirt. This prevents big organizations running many VMs of all kinds
+from adopting KubeVirt. To allow greater adoption of KubeVirt, this proposal
+aims to make these legacy OSes work on KubeVirt.
+
+For more information on the topic
+see [this thread on kubevirt-dev](https://groups.google.com/g/kubevirt-dev/c/1Ue4brPoc1g)
+and [this issue](https://bugzilla.redhat.com/show_bug.cgi?id=990418)
+
+## Goals
+
+<!--
+The desired outcome
+-->
+
+- A way of expressing that the 64-Bit PCI hole should be disabled is added to
+  the KubeVirt API in the most OS-independent way possible.
+- Legacy OSes like Windows XP or Server 2k3 can run on KubeVirt without using
+  `Sidecar` hooks.
+
+## Non Goals
+
+<!--
+Limitations to the scope of the design
+-->
+
+- Committing to a list of supported OSes in KubeVirt.
+- Adding OS specific configurables to the KubeVirt API.
```

Comment:
Is a `legacyWindowsCompat` annotation that magically disables PCI hole on the underlying domain not an OS specific configurable?

Reply:
Somewhat, yes. Not entirely OS specific, but I agree it is OS family specific.